src/global/GlobalState.js

```diff
@@ -0,0 +1,178 @@
+import React, { useState } from "react";
+import axios from "axios";
+import GlobalStateContext from "./GlobalStateContext";
+import { baseUrl } from "../constants/baseUrl";
+import { goToLoginPage } from "../router/Coordinator";
+
+const GlobalState = (props) => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  // Estado com todos os restaurantes
+  const [restaurants, setRestaurants] = useState([]);
+  // Estado com produtos do restaurante escolhido
+  const [restaurantProducts, setRestaurantProducts] = useState([]);
+  // Estado com detalhes do restaurante escolhido
+  const [restaurantDetails, setRestaurantDetails] = useState([]);
+  // Estado para armazenar dados do perfil
+  const [profileData, setProfileData] = useState({});
+  // Estado usado para filtrar por categoria
+  const [category, setCategory] = useState(0);
+  // Estado usado para filtrar por nome
+  const [filteredByName, setFilteredByName] = useState("");
+  // Variável com o token do localStorage
+  const token = localStorage.getItem("token");
+  // Variável com autorização para requisições
+  const auth = { headers: { auth: token } };
```

Comment:
Isolar a axiosConfig dessa forma, nesse caso, não é uma boa. Nos casos em que a configuração do headers necessita de um valor do localStorage, que pode mudar com o tempo, é melhor isolar tudo em uma função e chamar a função sempre que for usar a axiosConfig, dessa forma:
```

export const generateAxiosConfig = () =>{
     const axiosConfig = {
         headers: { auth: window.localStorage.getItem("token") }
     }

    return axiosConfig
}
```